ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/json/JsonArrayLength.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.json;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectReader;
+import io.confluent.ksql.function.FunctionCategory;
+import io.confluent.ksql.function.KsqlFunctionException;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import io.confluent.ksql.util.KsqlConstants;
+
+@UdfDescription(
+    name = "JSON_ARRAY_LENGTH",
+    category = FunctionCategory.JSON,
+    description = "Given a string, parses it as a JSON value and returns the length of the "
```

Comment:
Should the description explain the error that one gets with non-JSON input?

Reply:
Good point! Added in a8008bf.

Reply:
@jnh5y FYI I reverted that change because the function user never sees the exceptions; the function returns `null` in that case. See [discussion](https://github.com/confluentinc/ksql/pull/8550#issuecomment-1016954552) in the KLIP-59 as well as [this test](https://github.com/confluentinc/ksql/pull/8602/files#diff-3f2fe566cca5242303456caa8665d8b399cf731ff3118381aaccb0bc4fef8138R18) for more details.